src/include/connection.h

```diff
@@ -169,20 +169,25 @@ struct __wt_oligarch_manager {
 
 /*
  * WT_DISAGGREGATED_STORAGE --
- *      Configuration for disaggregated storage, which tells the Block Manager how to find remote
- *      object storage. This is a separate configuration from Oligarch tables.
+ *      Configuration and the current state for disaggregated storage, which tells the Block Manager
+ *      how to find remote object storage. This is a separate configuration from Oligarch tables.
  */
 struct __wt_disaggregated_storage {
     char *page_log;
     char *stable_prefix;
     char *storage_source;
 
+    uint64_t global_checkpoint_id; /* The ID of the currenty opened checkpoint. */
```

Comment:
should this be shared?

Reply:
I just made this a shared variable and expanded the comment.

Its updates are protected by the checkpoint lock (and yes, I was missing one), but then it's read from different threads without any protection, so I added the corresponding barriers.